kong/runloop/balancer.lua

```diff
@@ -505,7 +505,7 @@ local function check_target_history(upstream, balancer)
     end
   end
 
-  if last_equal_index == new_size and new_size > 0 then
+  if last_equal_index == new_size and new_size > 0 and new_size == old_size then
```

Comment:
Thank you! This is a better fix than the `new_size > 0` check that I added in 1.4.2. I believe it can replace it even:

```suggestion
  if last_equal_index == new_size and new_size == old_size then
```